script.js

```diff
@@ -1,4 +1,106 @@
+var playerResults = [];
+var clickCount = 0;
+var tempNum = [];
+var playerNumber = 1;
+var playerGroupNumber = [];
+
+// clickCount is to control the mode:
+// When clickCount equal to even number, it's "Dice-roll mode":
+// Generate 2 random numbers and show to players
+// When clickCount equal to odd number, it's "Push mode":
+// Finalise the number according to the order player choose, and push the number into array "playerResults"
+
+var getRandomIndex = function () {
+  // create a random number from zero to array length minus one.
+  // this number is in the range from the first
+  // index (zero) to the last index (array length minus one)
+  var randomIndex = Math.floor(Math.random() * 6) + 1;
+  return randomIndex;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+  var currentNumber;
+  var rollDice1;
+  var rollDice2;
+
+  // Dice-roll mode: clickCount equal to even number
+  if (clickCount % 2 == 0) {
```

Comment:
it might be easier to read if you just use `mode == 'dice-roll mode'` to show the different modes, because words are easier to understand than to use even/odd number, especially when the clickCount keeps increasing

Reply:
Noted